components/hydrator/internal/authnmappinghandler/token_data_cache.go

```diff
@@ -0,0 +1,71 @@
+package authnmappinghandler
+
+import (
+	"sync"
+	"time"
+)
+
+// Entry represents a single cache Entry
+type Entry struct {
+	TokenData TokenData
+	Taken     time.Time
+}
+
+// TokenDataCache is a local short-lived cache
+type TokenDataCache struct {
+	mutex sync.Mutex
+	ttl   time.Duration
+	data  map[string]map[string]Entry
```

Comment:
Could we simplify this and make it `map[string]Entry`, removing the double map? Holding the "token" -> Entry data. Because in the token we'll have embedded the issuer URL in the claims? That would simplify also the `PutTokenData`, `GetTokenData`, and `Cleanup` methods.